### DISCUSSION TITLE
"Transcribe your voice live" demo (sdk-js) could not connect websocket

### DISCUSSION THREAD

**Discussion by jerry2013** [deepgram/community#813] [General help] [answered]
### Which Deepgram product feature are you using?

Deepgram SDKs

### Details

the demo "mission/transcribe-your-voice-in-realtime" fails due to websocket connection failing.

BTW. your doc didn't mention putting token in ws protocol field.

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by lukeocodes** [comment, marked as answer]
Hey @jerry2013, where abouts are you? We have a known de-peering issue between some ISPs which could cause this. I've checked that mission and it is working for us (and I am in the UK, definitely not local to our services)

Re: subprotocol auth - it is an alternative method that we utilise in frontends, but we don't like to announce it as the primary solution. We do have a lot of frontend demos though, and the JS SDK supports subprotocol authentication, too.

**Comment by jerry2013** [comment]
interesting.

I'm in US, ISP is comcast in CT. I just tried from Rogers from Toronto Canada and got the same rejection.

Maybe it's the browser (chrome 126) - i'm going to try in nodejs.

**Comment by jerry2013** [comment]
ok. turns out, if, in the same browser, I manually change the wss to use `model=enhanced` then it worked (the demo uses `nova-2`)

the doc says `nova-2` is the shiniest new toy - does anyone know whether there's any limit placed on this model for paying customers?

**Reply by lukeocodes** [reply]
Can you let me have the request options/params you're getting the error on?

**Reply by jerry2013** [reply]
i'm just using what the demo has `wss://api.deepgram.com/v1/listen?encoding=ogg-opus&sample_rate=16000&smart_format=true&interim_results=true&model=enhanced&language=en-US` (changed model from `nova-2` to `enhanced` and it worked)

PS. I've emailed developers @ deepgram (i guess that's deepgram's support) to ask about why nova-2 doesn't work for my account.
